microsoft/utils/setup_AITL.py

```diff
@@ -0,0 +1,299 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+# This script is to create/update a custom role and grant to Service Principal for Partner in Production Env
+# Debug sample: .\Setup_AITL.py -r "AzureLSGDelegator_test" -p "AzureLinuxTestingService" -s "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+# User sample:  .\Setup_AITL.py
+
+import argparse
+import json
+import subprocess
+import sys
+import time
+
+
+def call_cmd(command: str) -> subprocess.CompletedProcess[str]:
+    # Run the command and capture the return code
+    try:
+        result = subprocess.run(
+            command,
+            shell=True,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+        return_code = result.returncode
+        print(f"Return Code: {return_code} for cmd: {command}")
+        # output = result.stdout
+        # print(f"Output:\n{output}") # can open for debug purpose
+        return result
+    except subprocess.CalledProcessError as e:
+        # If the command returns a non-zero exit code, you can handle the error here
+        print(f"Command failed with error: {e.stderr}")
+        sys.exit(1)
+    except Exception as e:
+        # Handle other exceptions here
+        print(f"An error occurred: {e}")
+        sys.exit(1)
+
+
+def _compare_str_list(str_list_1: list[str], str_list_2: list[str]) -> bool:
+    if len(str_list_1) != len(str_list_2):
+        return False
+
+    str_list_1.sort()
+    str_list_2.sort()
+    return str_list_1 == str_list_2
+
+
+def _compare_role_setting(role1_json: str, role2_json: str) -> bool:
```

Comment:
Don't use role1_list, role2_list. Use previous_definition, current_definition.

1. Avoid using unmeaningful names like 1, 2.
2. Avoid to use type in the name like _list, unless mutiple types on the same meaning.